### PR TITLE
ENG-4551 Insert CDS connector project into app-engine project

### PR DESCRIPTION
**Pull request by entgigi**

### CODE REVIEW

File: cds-plugin/pom.xml [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.entando.entando.plugins</groupId>
+            <artifactId>entando-plugin-jacms</artifactId>
+            <version>${project.version}</version>
+            <type>test-jar</type>
+            <scope>test</scope>
+        </dependency>
+        <!-- xkè ??? -->
```
Comment (zonia3000) [inline cds-plugin/pom.xml]: ?

File: cds-plugin/src/main/java/org/entando/entando/plugins/jpcds/aps/system/storage/CdsFileAttributeViewDto.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+@EqualsAndHashCode
+@ToString
+public class CdsFileAttributeViewDto implements Serializable {
+    
+    public static final String LONG_TIME_PARAM_NAME = "secs_since_epoch";
+    
+    private String name;
+	private Map<String,String> lastModifiedTime;
```
Comment (zonia3000) [inline cds-plugin/src/main/java/org/entando/entando/plugins/jpcds/aps/system/storage/CdsFileAttributeViewDto.java]: indentation

File: cds-plugin/src/main/java/org/entando/entando/plugins/jpcds/aps/system/storage/CdsRemoteCaller.java [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+import org.entando.entando.aps.system.services.storage.CdsActive;
+
+@Component
+@CdsActive(true)
+public class CdsRemoteCaller  {
+
+    private static final EntLogger logger = EntLogFactory.getSanitizedLogger(CdsRemoteCaller.class);
```
Comment (zonia3000) [inline cds-plugin/src/main/java/org/entando/entando/plugins/jpcds/aps/system/storage/CdsRemoteCaller.java]: Can we use sl4j?

File: cds-plugin/src/main/java/org/entando/entando/plugins/jpcds/aps/system/storage/CdsRemoteCaller.java [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+
+
+    private String extractToken(Optional<TenantConfig> config, boolean force) {
+         TenantConfig tc = config.orElse(null);
+         if(tc!= null){
+             return getTenantToken(tc, force);
+         } else {
+            return getPrimaryToken(force);
+         }
+         // FIXME doesn't work => return config.map(tc -> getTenantToken(tc,force)).orElse(getPrimaryToken(force));
```
Comment (zonia3000) [inline cds-plugin/src/main/java/org/entando/entando/plugins/jpcds/aps/system/storage/CdsRemoteCaller.java]: ?

File: cds-plugin/src/test/config/conf/security.properties [hunk 5]
```diff
@@ -0,0 +1,11 @@
+# Parameters for Argon2 algorithm encryption
```
Comment (eugeniosant) [inline cds-plugin/src/test/config/conf/security.properties]: This file can be removed (argon2 library removed)

File: cds-plugin/src/main/resources/spring/plugins/jpcds/aps/baseManagersConfig.xml [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+       xsi:schemaLocation="http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans.xsd 
+                        http://www.springframework.org/schema/util http://www.springframework.org/schema/util/spring-util-4.0.xsd
+			http://www.springframework.org/schema/context http://www.springframework.org/schema/context/spring-context.xsd 
+			http://www.springframework.org/schema/jee http://www.springframework.org/schema/jee/spring-jee.xsd 
+			http://www.springframework.org/schema/cache http://www.springframework.org/schema/cache/spring-cache.xsd">
+
+    <context:component-scan base-package="org.entando.entando.plugins.jpcds.aps.system.storage" />
+
+    <!--
+    <bean id="StorageManager" class="org.entando.entando.plugins.jpcds.aps.system.storage.CdsStorageManager">
```
Comment (eugeniosant) [inline cds-plugin/src/main/resources/spring/plugins/jpcds/aps/baseManagersConfig.xml]: Remove section

File: cds-plugin/src/test/config/resources/cms/documents/text_resource.txt [hunk 7]
```diff
@@ -0,0 +1,2 @@
+Entando is a low-code, component-based platform that opens the door for accelerated development of cloud Modern Applications that leverage the full potential of continuous integration/continuous delivery, devops, containers, and microservices. Entando unleashes developer creativity, freeing them to work lighter, faster and more efficiently.
```
Comment (eugeniosant) [inline cds-plugin/src/test/config/resources/cms/documents/text_resource.txt]: Is this file used in some test?

File: engine/src/main/java/org/entando/entando/aps/system/services/storage/LocalStorageManager.java [hunk 8]
```diff
@@ -24,25 +24,36 @@
...
+import org.springframework.stereotype.Service;
 
 import static org.entando.entando.aps.system.services.storage.StorageManagerUtil.isSamePath;
 
-public class LocalStorageManager implements IStorageManager {
+@Service("StorageManager")
+@CdsActive(false)
+public class LocalStorageManager implements IStorageManager, InitializingBean {
 
-	private static final EntLogger logger = EntLogFactory.getSanitizedLogger(LocalStorageManager.class);
```
Comment (eugeniosant) [inline engine/src/main/java/org/entando/entando/aps/system/services/storage/LocalStorageManager.java:32]: Why did the log class change?

File: cds-plugin/src/main/java/org/entando/entando/plugins/jpcds/aps/system/storage/CdsStorageManager.java [hunk 9]
```diff
@@ -0,0 +1,347 @@
...
+                .filter(StringUtils::isNotBlank)
+                .map(tenantManager::getConfig);
+    }
+    
+
+    private String validateAndReturnResourcePath(Optional<TenantConfig> config, String resourceRelativePath, boolean privateUrl) {
+        try {
+            String baseUrl = CdsUrlBuilder.builder()
+                    .url(CdsUrlUtils.fetchBaseUrl(config, configuration, privateUrl))
+                    .path(CdsUrlUtils.getInternalSection(privateUrl)) // << this is part of bash url because we want check path traversal!!
```
Comment (zonia3000) [inline cds-plugin/src/main/java/org/entando/entando/plugins/jpcds/aps/system/storage/CdsStorageManager.java]: bash -> base

File: cds-plugin/src/main/resources/spring/plugins/jpcds/aps/baseManagersConfig.xml [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+        <property name="enabled" value="${cds.enabled:false}" />
+        <property name="cdsPublicUrl" value="${cds.public.url:https://cds.entando.org}" />
+        <property name="cdpPrivateUrl" value="${cds.private.url:https://cds.entando.org}" />
+        <property name="cdsPath" value="${cds.path:/api/v1}" />
+        <property name="kcAuthUrl" value="${keycloak.auth.url:http://localhost:8081/auth}" />
+        <property name="kcRealm" value="${keycloak.realm:entando}" />
+        <property name="kcClientId" value="${keycloak.client.id:entando-app}" />
+        <property name="kcClientSecret" value="${keycloak.client.secret:}" />
+    </bean>
+    -->
```
Comment (zonia3000) [inline cds-plugin/src/main/resources/spring/plugins/jpcds/aps/baseManagersConfig.xml]: Can we remove this commented section?

File: engine/src/main/resources/spring/baseSystemConfig.xml [hunk 11]
```diff
@@ -28,6 +30,7 @@
         </property>
     </bean>
 
+    <!--
```
Comment (zonia3000) [inline engine/src/main/resources/spring/baseSystemConfig.xml]: Can we remove this commented section?

File: cds-plugin/src/test/resources/document/architecture.txt [hunk 12]
```diff
@@ -0,0 +1,10 @@
+The Entando platform includes the following major elements:
+
+Entando App Engine: assemble micro frontends & microservices and compose them on a page.
```
Comment (eugeniosant) [inline cds-plugin/src/test/resources/document/architecture.txt]: Is this file used in some test?

**Review by eugeniosant — CHANGES_REQUESTED** [review]

File: engine/src/main/java/org/entando/entando/aps/system/services/storage/LocalStorageManager.java (hunk 8 above)
Comment (entgigi) [reply engine/src/main/java/org/entando/entando/aps/system/services/storage/LocalStorageManager.java:32]: aren't we using Slf4j everywhere?

File: cds-plugin/src/test/config/resources/cms/documents/text_resource.txt (hunk 7 above)
Comment (entgigi) [reply cds-plugin/src/test/config/resources/cms/documents/text_resource.txt]: removed

File: cds-plugin/src/main/resources/spring/plugins/jpcds/aps/baseManagersConfig.xml (hunk 6 above)
Comment (entgigi) [reply cds-plugin/src/main/resources/spring/plugins/jpcds/aps/baseManagersConfig.xml]: removed

File: cds-plugin/src/test/config/conf/security.properties (hunk 5 above)
Comment (entgigi) [reply cds-plugin/src/test/config/conf/security.properties]: removed

File: cds-plugin/src/test/resources/document/architecture.txt (hunk 12 above)
Comment (entgigi) [reply cds-plugin/src/test/resources/document/architecture.txt]: removed

File: cds-plugin/pom.xml (hunk 1 above)
Comment (entgigi) [reply cds-plugin/pom.xml]: removed

File: cds-plugin/src/main/java/org/entando/entando/plugins/jpcds/aps/system/storage/CdsRemoteCaller.java (hunk 3 above)
Comment (entgigi) [reply cds-plugin/src/main/java/org/entando/entando/plugins/jpcds/aps/system/storage/CdsRemoteCaller.java]: yes for sure, that  is my bad

File: cds-plugin/src/main/java/org/entando/entando/plugins/jpcds/aps/system/storage/CdsRemoteCaller.java (hunk 4 above)
Comment (entgigi) [reply cds-plugin/src/main/java/org/entando/entando/plugins/jpcds/aps/system/storage/CdsRemoteCaller.java]: It is something I want to talk with you about it ... I mentioned it to you days ago ...

File: cds-plugin/src/main/resources/spring/plugins/jpcds/aps/baseManagersConfig.xml (hunk 10 above)
Comment (entgigi) [reply cds-plugin/src/main/resources/spring/plugins/jpcds/aps/baseManagersConfig.xml]: already removed

File: engine/src/main/resources/spring/baseSystemConfig.xml (hunk 11 above)
Comment (entgigi) [reply engine/src/main/resources/spring/baseSystemConfig.xml]: removed

File: cds-plugin/src/main/java/org/entando/entando/plugins/jpcds/aps/system/storage/CdsFileAttributeViewDto.java (hunk 2 above)
Comment (entgigi) [reply cds-plugin/src/main/java/org/entando/entando/plugins/jpcds/aps/system/storage/CdsFileAttributeViewDto.java]: reformatted all file

File: engine/src/main/java/org/entando/entando/aps/system/services/storage/LocalStorageManager.java (hunk 8 above)
Comment (entgigi) [reply engine/src/main/java/org/entando/entando/aps/system/services/storage/LocalStorageManager.java:32]: talked with Eugenio and Sonia face2face we keep slf4j

File: cds-plugin/src/main/java/org/entando/entando/plugins/jpcds/aps/system/storage/CdsStorageManager.java (hunk 9 above)
Comment (entgigi) [reply cds-plugin/src/main/java/org/entando/entando/plugins/jpcds/aps/system/storage/CdsStorageManager.java]: done

File: cds-plugin/src/main/java/org/entando/entando/plugins/jpcds/aps/system/storage/CdsRemoteCaller.java (hunk 4 above)
Comment (entgigi) [reply cds-plugin/src/main/java/org/entando/entando/plugins/jpcds/aps/system/storage/CdsRemoteCaller.java]: changed as you suggested tnx!

**Review by zonia3000 — APPROVED** [review]

**Review by eugeniosant — APPROVED** [review]

**Pull request merged**